crates/air-lib/air-parser/src/ast/values.rs

```diff
@@ -86,3 +86,10 @@ pub struct StreamMap<'i> {
     pub name: &'i str,
     pub position: AirPos,
 }
+
+/// An error wrapper with an optional lens.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct InstructionErrorAST<'lens> {
+    #[serde(borrow)]
+    pub lens_option: Option<LambdaAST<'lens>>,
```

Comment:
let's call it just `lens`.

Reply:
There will be shadowing in many places b/c lens_option contains lens inside. I agree with Ivan that shadowing is bad for reading.

Reply:
Can you show exactly, please?

Generally, "_option" and other type designating suffixes are not used in Rust because of real strong typing.

Reply:
The latest changes removed this shadowing from the code.

Reply:
I found the example. It is in display_error() in crates/air-lib/air-parser/src/ast/instruction_arguments/traits.rs

Reply:
Such shadowing (of a related value) is totally fine.